stdlib/std.jsonnet

```diff
@@ -1066,45 +1069,46 @@ limitations under the License.
     std.join('', std.map(function(b) std.char(b), bytes)),
 
   // Quicksort
-  sort(arr)::
+  sort(arr, keyF=id)::
     local l = std.length(arr);
     if std.length(arr) == 0 then
       []
     else
-      local pivot = arr[0];
+      local pivot = keyF(arr[0]);
       local rest = std.makeArray(l - 1, function(i) arr[i + 1]);
-      local left = std.filter(function(x) x <= pivot, rest);
-      local right = std.filter(function(x) x > pivot, rest);
-      std.sort(left) + [pivot] + std.sort(right),
+      local left = std.filter(function(x) keyF(x) < pivot, rest);
+      local right = std.filter(function(x) keyF(x) >= pivot, rest);
```

Comment:
When merging sets, often you have preference which elements to keep, therefore sorting need to be stable, then values coming first are "winning". This change seems to me enough to make quicksort stable, but please have a look

Reply:
I think it works.

Proof:
It works for length 0. Not let's assume it's stable for all lengths < n (*). Filter doesn't change the relative order of elements < pivot. By (*) after sorting, the order of equal elements in this subarray is maintained. Similarly for elements >= pivot. Pivot is the first element of the array, so you know that it goes before any elements >= pivot.  So the order of equal elements is maintained.